### ISSUE TITLE
关于get_std_reins_eva02_large的配置问题

### ISSUE THREAD

**Issue by V1perQAQ0821** [w1oves/Rein#81]
您好，非常感谢您提供的代码。我计划在我的模型中集成 reins_eva_02，并参考了 simple_reins 中的 reins_dinov2.py 和 reins_clip.py 文件，编写了 get_std_reins_eva02_large 函数。然而，代码未能正常运行。您能否提供 get_std_reins_eva02_large 函数的正确配置方法？对此，我将不胜感激。

    class ReinsEVA2(EVA2):
    def __init__(self, reins_config=None, **kwargs):
        super().__init__(**kwargs)
        self.reins: Reins = MODELS.build(reins_config)

    def forward_features(self, x):
        B, C, H, W = x.shape
        x, (Hp, Wp) = self.patch_embed(x)
        batch_size, seq_len, _ = x.size()

        cls_tokens = self.cls_token.expand(
            batch_size, -1, -1
        )  # stole cls_tokens impl from Phil Wang, thanks
        x = torch.cat((cls_tokens, x), dim=1)
        if self.pos_embed is not None:
            x = x + self.pos_embed
        x = self.pos_drop(x)

        rel_pos_bias = self.rel_pos_bias() if self.rel_pos_bias is not None else None
        features = []
        for i, blk in enumerate(self.blocks):
            if self.use_checkpoint:
                x = checkpoint.checkpoint(blk, x, rel_pos_bias)
            else:
                x = blk(x, rel_pos_bias)
            x = self.reins.forward(
                x,
                i,
                batch_first=True,
                has_cls_token=True,
            )
            if i in self.out_indices:
                xp = x[:, 1:, :].permute(0, 2, 1).reshape(B, -1, Hp, Wp)
                features.append(xp.contiguous())
        features[0] = F.interpolate(
            features[0], scale_factor=4, mode="bilinear", align_corners=False
        )
        features[1] = F.interpolate(
            features[1], scale_factor=2, mode="bilinear", align_corners=False
        )
        features[3] = F.interpolate(
            features[3], scale_factor=0.5, mode="bilinear", align_corners=False
        )
        return self.reins.return_auto(features)

    def train(self, mode: bool = True):
        if not mode:
            return super().train(mode)
        set_requires_grad(self, ["reins"])
        set_train(self, ["reins"])

    def get_std_reins_eva02_large(checkpoint_file=None):
    # Reins 配置
    reins_config = dict(
        token_length=100,    # 跨模态输入的 token 长度
        embed_dims=1024,     # 嵌入维度
        num_layers=24,       # Reins 层数
        patch_size=16,       # Patch 大小
        lora_dim=16,         # LoRA 降维参数
    )

    # EVA02-L 配置
    return EVA2(
        reins_config=reins_config,  # 跨模态配置

        # 主干模型配置
        img_size=512,  # 输入图像尺寸
        patch_size=16,  # 图像块大小
        in_chans=3,  # 输入通道数（RGB图像）
        embed_dim=1024,  # 嵌入维度
        depth=24,  # Transformer 层数
        num_heads=16,  # 注意力头数量
        mlp_ratio=2.6667,  # FFN 扩展比例，贴合实际配置
        qkv_bias=True,  # 启用QKV偏置
        drop_path_rate=0.2,  # 路径丢弃率，增强模型鲁棒性
        norm_layer=dict(eps=1e-6, requires_grad=True, type="LN"),  # 层归一化配置
        out_indices=[7, 11, 15, 23],  # 多尺度输出索引

        # 初始化与训练优化配置
        init_values=None,  # 不使用层初始化值
        intp_freq=True,  # 插值频率启用
        pretrained=checkpoint_file,  # 预训练权重文件路径
        pt_hw_seq_len=16,  # 图像块序列长度

        # 高级特性启用
        use_checkpoint=False,  # 不启用检查点保存
        use_abs_pos_emb=True,  # 启用绝对位置嵌入
        use_rel_pos_bias=False,  # 不启用相对位置偏置
        use_shared_rel_pos_bias=False,  # 不共享相对位置偏置
        rope=True,  # 启用旋转位置编码
        subln=True,  # 启用子层归一化
        xattn=True  # 启用交叉注意力
    )
